galaxy_ng/app/api/ui/urls.py

```diff
@@ -180,6 +180,26 @@
     ),
 ]
 
+public_key_paths = [
```

Comment:
This set of paths and their corresponding views don't fit any of our standards.

1. Setting `base` as a url kwarg when there are only two options is confusing and makes the API way harder to navigate. It should have a separate set of paths for `repository` and `signing_service`
2. `PublicKeyView` is a viewset that's masquerading as a view. This should be rewritten as a viewset that has a `list` function for listing out the public keys and a `retrieve` function for retrieving a specific key. `api.base.ViewSet` should work for this since you've got a really weird set of querysets for this view and the base `ViewSet` doesn't provide any magic around pagination and viewsets
3. `PublicKeyView` doesn't declare any serializers or schemas, so it will break the API spec.

For the router you can do something like this. You could also make the file extension explicit as well, like you already have. 

```
[
   path(
        "",
        views.PublicKeyViewset.as_view({"get":"list"}),
        name='public-key',
    ),
    path(
        "repository/<path:file>",
        views.PublicKeyViewset.as_view({"get":"retrieve_repository"}),
        name='public-key-detail-key',
    ),
    path(
        "signing_service/<path:file>",
        views.PublicKeyViewset.as_view({"get":"retrieve_signing_service"}),
        name='public-key-detail-fingerprint',
    )
]
```

Reply:
@newswangerd actually I will be removing this endpoint, just learned that we don't need to expose the public_key fingerprint for uploads anymore because the upload viewset can fetch it from signature.

Also, signing service public_key is already available on pulp/signing_Service/ API